python-avd/pyavd/_eos_designs/structured_config/network_services/utils.py

```diff
@@ -330,7 +330,7 @@ def get_vrf_rd(self: AvdStructuredConfigNetworkServices, vrf: EosDesigns._Dynami
         rd_override = vrf.rd_override
 
         if rd_override is not None:
-            if ":" in rd_override:
+            if isinstance(rd_override, str) and (":" in rd_override or rd_override == "auto"):
```

Comment:
We already know that the value is either a str or None.
```suggestion
            if ":" in rd_override or rd_override == "auto":
```